p2p/src/test/kotlin/maru/p2p/discovery/MaruDiscoveryServiceTest.kt

```diff
@@ -236,47 +216,39 @@ class MaruDiscoveryServiceTest {
               P2P.Discovery(
                 port = PORT6,
                 bootnodes = listOf(enrString),
-                refreshInterval = 5.seconds,
+                refreshInterval = 10.seconds,
               ),
           ),
         forkIdHashProvider = forkIdHashProvider,
-        metricsSystem = NoOpMetricsSystem(),
       )
 
     try {
       bootnode.start()
       discoveryService2.start()
       discoveryService3.start()
 
-      awaitPeerFound(bootnode, discoveryService2.getLocalNodeRecord().nodeId)
-      awaitPeerFound(discoveryService2, discoveryService3.getLocalNodeRecord().nodeId)
-      awaitPeerFound(discoveryService3, discoveryService2.getLocalNodeRecord().nodeId)
-      awaitPeerFound(bootnode, discoveryService3.getLocalNodeRecord().nodeId)
+      // make sure services are started and bootnodes have been pinged
+      sleep(3000)
+
+      discoveryService2
+        .searchForPeers()
+        .thenAccept { foundPeers ->
+          assertThat(foundPeers.any { it.nodeId == bootnode.getLocalNodeRecord().nodeId }).isTrue
+          assertThat(foundPeers.any { it.nodeId == discoveryService3.getLocalNodeRecord().nodeId }).isTrue
+        }.join()
+      discoveryService3
+        .searchForPeers()
+        .thenAccept { foundPeers ->
+          assertThat(foundPeers.any { it.nodeId == bootnode.getLocalNodeRecord().nodeId }).isTrue
```

Comment:
Minor: You can factor these assertions or the `searchForPeers` call with assertions to a function to make the tests a bit more readable

Reply:
done